File: WinPaletter/Store/Store_SearchFilter.vb
Public Class Store_SearchFilter

    Private Sub Store_SearchFilter_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        LoadLanguage
        ApplyStyle(Me)
        Icon = Store.Icon

        XenonCheckBox1.Checked = My.Settings.Store.Search_ThemeNames
        XenonCheckBox2.Checked = My.Settings.Store.Search_AuthorsNames
        XenonCheckBox3.Checked = My.Settings.Store.Search_Descriptions

    End Sub

    Private Sub XenonButton1_Click(sender As Object, e As EventArgs) Handles XenonButton1.Click
        My.Settings.Store.Search_ThemeNames = XenonCheckBox1.Checked
        My.Settings.Store.Search_AuthorsNames = XenonCheckBox2.Checked
        My.Settings.Store.Search_Descriptions = XenonCheckBox3.Checked
        My.Settings.Store.Save()
        Close()
    End Sub

    Private Sub XenonButton7_Click(sender As Object, e As EventArgs) Handles XenonButton7.Click
        Close()
    End Sub
End Class

File: WinPaletter/Store/Store_SearchFilter.cs
using FluentTransitions;
using System;
using System.Drawing;
using System.Windows.Forms;

namespace WinPaletter
{
    /// <summary>
    /// Search filter form for the store.
    /// </summary>
    public partial class Store_SearchFilter
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Store_SearchFilter"/> class.
        /// </summary>
        public Store_SearchFilter()
        {
            InitializeComponent();
        }

        Size targetSize;

        private void Store_SearchFilter_Load(object sender, EventArgs e)
        {
            CheckBox1.Checked = Program.Settings.Store.Search_ThemeNames;
            CheckBox2.Checked = Program.Settings.Store.Search_AuthorsNames;
            CheckBox3.Checked = Program.Settings.Store.Search_Descriptions;

             // Animate the form.
            Transition.With(this, nameof(Height), targetSize.Height).CriticalDamp(TimeSpan.FromMilliseconds(Program.AnimationDuration * 0.6));
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            Program.Settings.Store.Search_ThemeNames = CheckBox1.Checked;
            Program.Settings.Store.Search_AuthorsNames = CheckBox2.Checked;
            Program.Settings.Store.Search_Descriptions = CheckBox3.Checked;
            Program.Settings.Store.Save();
            Close();
        }

        public DialogResult ShowDialog(Size parentButtonSize, Point parentButtonLocation)
        {
            targetSize = Size;

            Size = new(Width, 1);
            Location = parentButtonLocation;
            Left -= Width - parentButtonSize.Width;

            return this.ShowDialog();
        }

        private void Button7_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}